### PR TITLE
Inspector polish

### PR DESCRIPTION
**Pull request by jacob314**
Correctly display which inspector tree nodes are expandable.
Computer inspector width based on max tree depth.
Switch to simpler ResizeObserver scheme.

There was a bug where the summary tree always showed expand-collapse icons even when they weren't needed. Fixed the bug and updated the goldens to reflect the correct state.
Fixed bug where the inspector tree width didn't update to reflect very deep trees.
Example with track widget creation off showing a deep tree is fine:
Screen Shot 2019-05-02 at 12 16 52 AM

### CODE REVIEW

File: packages/devtools/lib/src/ui/viewport_canvas.dart [hunk 1]
```diff
@@ -167,47 +173,14 @@ class ViewportCanvas extends Object with SetStateMixin {
...
-    _visibilityObserver = IntersectionObserver(
-        allowInterop(_onVisibilityChange), {'root': _element.element});
-    for (var spacer in spacers) {
-      _visibilityObserver.observe(spacer.element);
-    }
+    // TODO(jacobr): clean this code up when
+    // https://github.com/dart-lang/html/issues/104 is fixed.
+    final observer = ResizeObserver(allowInterop((List<dynamic> entries, _) {
+      // XXX _scheduleRebuild();
+      _hasPendingRebuild = false;
```
Comment (kenzieschmoll) [inline packages/devtools/lib/src/ui/viewport_canvas.dart]: I'm not seeing where `_hasPendingRebuild` is set to true?

File: packages/devtools/lib/src/ui/viewport_canvas.dart [hunk 2]
```diff
@@ -167,47 +173,14 @@ class ViewportCanvas extends Object with SetStateMixin {
...
-    // https://github.com/dart-lang/sdk/issues/35484 is fixed.
-    _visibilityObserver = IntersectionObserver(
-        allowInterop(_onVisibilityChange), {'root': _element.element});
-    for (var spacer in spacers) {
-      _visibilityObserver.observe(spacer.element);
-    }
+    // TODO(jacobr): clean this code up when
+    // https://github.com/dart-lang/html/issues/104 is fixed.
+    final observer = ResizeObserver(allowInterop((List<dynamic> entries, _) {
+      // XXX _scheduleRebuild();
```
Comment (kenzieschmoll) [inline packages/devtools/lib/src/ui/viewport_canvas.dart]: nit: can we remove this comment?

File: packages/devtools/lib/src/ui/viewport_canvas.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools/lib/src/ui/viewport_canvas.dart]: missed setting this back before I commit. XXX comments are supposed to a reminder to me to fix before committing.
switched to just calling _scheduleRebuild(). I was trying to be less async to handle a case from the perf canvas view but that will be a separate CL.

File: packages/devtools/lib/src/ui/viewport_canvas.dart (hunk 2 above)
Comment (jacob314) [reply packages/devtools/lib/src/ui/viewport_canvas.dart]: yes. Switched to that code. Any time you see XXX in my code it means I missed fixing something before I send out for review.

**Comment by jacob314** [comment]
ptal

**Pull request merged**